src/simtools/ray_tracing/incident_angles.py

```diff
@@ -0,0 +1,592 @@
+        self._run_script(run_script, log_file)
+
+        data = self._compute_incidence_angles_from_imaging_list(photons_file)
+        self.results = QTable()
+        self.results["angle_incidence_focal"] = data["angle_incidence_focal_deg"] * u.deg
+        if self.calculate_primary_secondary_angles:
+            if "angle_incidence_primary_deg" in data:
+                self.results["angle_incidence_primary"] = (
+                    data["angle_incidence_primary_deg"] * u.deg
+                )
+            if "angle_incidence_secondary_deg" in data:
+                self.results["angle_incidence_secondary"] = (
+                    data["angle_incidence_secondary_deg"] * u.deg
+                )
+            if "primary_hit_radius_m" in data:
+                self.results["primary_hit_radius"] = data["primary_hit_radius_m"] * u.m
+            if "secondary_hit_radius_m" in data:
+                self.results["secondary_hit_radius"] = data["secondary_hit_radius_m"] * u.m
+            if "primary_hit_x_m" in data and "primary_hit_y_m" in data:
+                self.results["primary_hit_x"] = data["primary_hit_x_m"] * u.m
+                self.results["primary_hit_y"] = data["primary_hit_y_m"] * u.m
+            if "secondary_hit_x_m" in data and "secondary_hit_y_m" in data:
+                self.results["secondary_hit_x"] = data["secondary_hit_x_m"] * u.m
+                self.results["secondary_hit_y"] = data["secondary_hit_y_m"] * u.m
+
+        self._save_results()
+        return self.results
+
+    def run_for_offsets(self, offsets):
+        """Run the simulation for multiple off-axis angles.
+
+        For each off-axis angle provided, run a full simulation, labeling output files
+        accordingly. Returns a mapping from off-axis angle (deg) to the resulting QTable.
+        """
+        results_by_offset = {}
+        base_off = self.rt_params.get("off_axis_angle", 0.0 * u.deg)
+
+        for off in offsets:
+            self.rt_params["off_axis_angle"] = float(off) * u.deg
+            self.logger.info(f"Running for off-axis angle {off:g} deg with label {self.label}")
+            tbl = self.run()
+            results_by_offset[float(off)] = tbl.copy()
+
+        self.rt_params["off_axis_angle"] = base_off
+        return results_by_offset
+
+    def _prepare_psf_io_files(self):
+        """Prepare photons, stars, and log file paths for a PSF-style incident angle simulation."""
+        suffix = self._label_suffix()
+        photons_file = self.photons_dir / f"incident_angles_photons_{suffix}.lis"
+        stars_file = self.photons_dir / f"incident_angles_stars_{suffix}.lis"
+        log_file = self.logs_dir / f"incident_angles_{suffix}.log"
+
+        if photons_file.exists():
+            try:
+                photons_file.unlink()
+            except OSError as err:
+                self.logger.warning(f"Failed to remove existing photons file {photons_file}: {err}")
```

Comment:
Why is this a warning only? Wouldn't that be an indication of going something more wrong when we can't delete the file? My suspicion is that this should be an error.